agent/command/s3_put.go

```diff
@@ -85,10 +87,14 @@ type s3put struct {
 	// for missing files.
 	Optional string `mapstructure:"optional" plugin:"expand"`
 
+	// NoOverwrite, when set to true, will not upload files if they already exist in s3
+	NoOverwrite string `mapstructure:"no_overwrite" plugin:"expand"`
```

Comment:
I don't think that's true, because I used ` no_overwrite: true` when I was testing and it works. I think it just automatically parses the yaml as strings. I might be wrong though.

Reply:
I modeled this after optional, which is also a boolean.